compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -632,7 +628,14 @@ object Contexts {
     def setRun(run: Run): this.type = updateStore(runLoc, run)
     def setProfiler(profiler: Profiler): this.type = updateStore(profilerLoc, profiler)
     def setNotNullInfos(notNullInfos: List[NotNullInfo]): this.type = updateStore(notNullInfosLoc, notNullInfos)
-    def setImportInfo(importInfo: ImportInfo): this.type = updateStore(importInfoLoc, importInfo)
+    def setImportInfo(importInfo: ImportInfo): this.type =
+      importInfo.mentionsFeature(nme.unsafeNulls) match
+        case Some(true) =>
+          setMode(this.mode &~ Mode.SafeNulls)
+        case Some(false) if ctx.settings.YexplicitNulls.value =>
```

Comment:
Also need `&& !Feature.enabledBySetting(nme.unsafeNulls)`

Reply:
I think not. The initial setting of `unsafeNulls` is irrelevant here.

Reply:
If we don't have the test and `unsafeNulls` is enabled globally, the `SafeNulls` will be added to the context here, which is not expected.

Reply:
No, I think that _is_ the expected behavior. A local (un-)import overrides an outer import and a global setting.

Reply:
I think you are right. We should allow local imports to overrides a global setting.